product-service/handlers/createProduct.ts

```diff
@@ -0,0 +1,54 @@
+import { APIGatewayProxyHandler } from "aws-lambda";
+import { HTTPMethods, getHeaders } from "../utils/response-headers";
+import { Client } from "pg";
+import dbOptions from "../db/options";
+
+export const createProduct: APIGatewayProxyHandler = async (
```

Comment:
this is not a transaction-based product creation.

product should not be created in DB if Stocks creation fails.

There are no such conditions in code.

Better to follow pattern: `BEGIN -> COMMIT OR REVERT`

Reply:
Transaction-based creation roughly will look like this:
`
try {
 client.query('BEGIN');
 //same magic of creating product and stocks as you have
 client.query('COMMIT');
} catch () {client.query('ROLLBACK')}
`

Reply:
fixed, thank you